src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -109,31 +113,25 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
-
-    CompletableFuture<Result<InstanceRequestRelatedRecords>> sortedAvailableItemsFuture =
-        itemsListsFuture.thenApply(r -> r.next(RequestByInstanceIdResource::rankItemsByMatchingServicePoint));
-    CompletableFuture<Result<Map<Item, DateTime>>> loanItemsFuture =
-        itemsListsFuture.thenCompose(r -> r.after(relatedRecords -> getLoanItems(relatedRecords, clients)));
-    CompletableFuture<Result<Map<Item, Integer>>> requestQueuesFuture =
-        itemsListsFuture.thenCompose(r -> r.after(relatedRecords -> getRequestQueues(relatedRecords, clients)));
-
-    CompletableFuture.allOf(sortedAvailableItemsFuture, loanItemsFuture, requestQueuesFuture)
-                     .thenApply( x -> combineWithUnavailableItems(loanItemsFuture.join(), requestQueuesFuture.join(), requestRelatedRecords))
-                     .thenApply( r -> r.next(RequestByInstanceIdResource::instanceToItemRequests))
-                     .thenCompose( r -> r.after( requests -> placeRequests(requests, clients)))
-                     .thenApply(r -> r.map(RequestAndRelatedRecords::getRequest))
-                     .thenApply(r -> r.map(new RequestRepresentation()::extendedRepresentation))
-                     .thenApply(CreatedJsonResponseResult::from)
-                     .thenAccept(result -> result.writeTo(routingContext.response()))
-                     .exceptionally( err -> {
-                         String reason = "Error processing title-level request";
-                         log.error(reason, err);
-                         ServerErrorResponse.internalError(routingContext.response(), reason);
-                         return null;
-                      });
+    finder.getItemsByInstanceId(requestByInstanceIdRequest.getInstanceId())
+          .thenCompose(r -> r.after(items -> getRequestQueues(items,requestRelatedRecords, clients)))
+          .thenApply(r -> r.next(items -> validateRequester(items, requestByInstanceIdRequest.getRequesterId())))
+          .thenApply(r -> r.next(items -> segregateItemsList(items, requestRelatedRecords)))
+          .thenApply(r -> r.next(RequestByInstanceIdResource::rankItemsByMatchingServicePoint))
+          .thenCompose(r -> r.after(relatedRecords -> getLoanItems(relatedRecords, clients)))
+          .thenApply( r -> r.next(loanItems -> combineWithUnavailableItems(loanItems, requestRelatedRecords)))
+          .thenApply( r -> r.next(RequestByInstanceIdResource::instanceToItemRequests))
+          .thenCompose( r -> r.after( requests -> placeRequests(requests, clients)))
+          .thenApply(r -> r.map(RequestAndRelatedRecords::getRequest))
+          .thenApply(r -> r.map(new RequestRepresentation()::extendedRepresentation))
+          .thenApply(CreatedJsonResponseResult::from)
+          .thenAccept(result -> result.writeTo(routingContext.response()))
+          .exceptionally( err -> {
+             String reason = "Error processing title-level request";
```

Comment:
When we come to consensus on title vs instance, we should update this message as well. I believe we should stick to FOLIO terms, instance in this case, but will leave the decision to the PO.